src/util/curves.ts

```diff
@@ -0,0 +1,136 @@
+
+// Factorial
+function fact(n: number): number {
+  let x = 1;
+  for (let i = 2; i <= n; i++) {
+    x *= i;
+  }
+  return x;
+}
+
+// Binomial Coefficients
+function bincoeff(n: number, k: number): number {
+  if (k == 0) {
+    return 1;
+  }
+  return fact(n) / (fact(k) * (fact(n - k)));
```

Comment:
Please don't calculate n-choose-k with the factorial-based definition, which is only really used in mathematical proofs. Factorials blow up very quickly.

Simplify the useless terms in the factorials instead: https://en.wikipedia.org/wiki/Binomial_coefficient#Multiplicative_formula